src/SocketContext.jsx

```diff
@@ -0,0 +1,76 @@
+import React, { createContext, useContext, useEffect, useState } from 'react';
+import io from 'socket.io-client';
+import { useAuth } from './AuthProvider';
+import NotificationModal from '../src/components/common/notificationModal';
+
+const SocketContext = createContext();
+
+export const SocketProvider = ({ children }) => {
+  const { userData } = useAuth();
+  const [socket, setSocket] = useState(null);
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [notificationMessage, setNotificationMessage] = useState('');
+
+  useEffect(() => {
+    if (!userData) {
+      console.error('No user data found');
+      return;
+    }
+
+    const userId = userData._id;
+
+    const socketIo = io('http://localhost:8000', {
```

Comment:
This is fine but to give it more flexibility when working between dev & prod this would help:
`process.env.REACT_APP_SOCKET_URL || 'http://localhost:8000'`